pkg/api/apis/operators/v1alpha1/clusterserviceversion_types.go

```diff
@@ -77,62 +78,73 @@ func (d *StrategyDetailsDeployment) GetStrategyName() string {
-	Path         string           `json:"path"`
-	DisplayName  string           `json:"displayName,omitempty"`
-	Description  string           `json:"description,omitempty"`
+	Path        string `json:"path"`
+	DisplayName string `json:"displayName,omitempty"`
+	Description string `json:"description,omitempty"`
+	// +listType=set
 	XDescriptors []string         `json:"x-descriptors,omitempty"`
 	Value        *json.RawMessage `json:"value,omitempty"`
 }
 
 // SpecDescriptor describes a field in a spec block of a CRD so that OLM can consume it
 // +k8s:openapi-gen=true
 type SpecDescriptor struct {
-	Path         string           `json:"path"`
-	DisplayName  string           `json:"displayName,omitempty"`
-	Description  string           `json:"description,omitempty"`
+	Path        string `json:"path"`
+	DisplayName string `json:"displayName,omitempty"`
+	Description string `json:"description,omitempty"`
+	// +listType=set
 	XDescriptors []string         `json:"x-descriptors,omitempty"`
 	Value        *json.RawMessage `json:"value,omitempty"`
 }
 
 // ActionDescriptor describes a declarative action that can be performed on a custom resource instance
 // +k8s:openapi-gen=true
 type ActionDescriptor struct {
-	Path         string           `json:"path"`
-	DisplayName  string           `json:"displayName,omitempty"`
-	Description  string           `json:"description,omitempty"`
+	Path        string `json:"path"`
+	DisplayName string `json:"displayName,omitempty"`
+	Description string `json:"description,omitempty"`
+	// +listType=set
 	XDescriptors []string         `json:"x-descriptors,omitempty"`
 	Value        *json.RawMessage `json:"value,omitempty"`
 }
 
 // CRDDescription provides details to OLM about the CRDs
 // +k8s:openapi-gen=true
 type CRDDescription struct {
-	Name              string                 `json:"name"`
-	Version           string                 `json:"version"`
-	Kind              string                 `json:"kind"`
-	DisplayName       string                 `json:"displayName,omitempty"`
-	Description       string                 `json:"description,omitempty"`
-	Resources         []APIResourceReference `json:"resources,omitempty"`
-	StatusDescriptors []StatusDescriptor     `json:"statusDescriptors,omitempty"`
-	SpecDescriptors   []SpecDescriptor       `json:"specDescriptors,omitempty"`
-	ActionDescriptor  []ActionDescriptor     `json:"actionDescriptors,omitempty"`
+	Name        string `json:"name"`
+	Version     string `json:"version"`
+	Kind        string `json:"kind"`
+	DisplayName string `json:"displayName,omitempty"`
+	Description string `json:"description,omitempty"`
+	// +listType=set
+	Resources []APIResourceReference `json:"resources,omitempty"`
+	// +listType=set
+	StatusDescriptors []StatusDescriptor `json:"statusDescriptors,omitempty"`
+	// +listType=set
+	SpecDescriptors []SpecDescriptor `json:"specDescriptors,omitempty"`
+	// +listType=set
+	ActionDescriptors []ActionDescriptor `json:"actionDescriptors,omitempty"`
```

Comment:
Although this is correct grammatically, this is a breaking API Change.

The change to the json tag will result in an updated CRD that does not include the `ActionDescriptor` field, in turn breaking any CRs on cluster that use the field.

Reply:
@awgreene I just changed the internal struct and not the json tag. Please see the changes to the CRDs in this [commit](https://github.com/operator-framework/operator-lifecycle-manager/pull/1370/commits/57b580e57e88f197617c7557d974057f6d9695a5). None of the fields are changed, only comments are updated. Note that `actionDescriptors` in the CRD remains `actionDescriptors`.

Reply:
Ah - I see - the json tag was always `actionDescriptors`. Got it.